react/components/NewChannelForm/index.js

```diff
@@ -18,17 +18,29 @@ class NewChannelForm extends Component {
   static propTypes = {
     createChannel: PropTypes.func.isRequired,
     onClose: PropTypes.func.isRequired,
+    authorType: PropTypes.oneOf(['USER', 'GROUP']),
+    group_id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
   }
 
-  state = {
-    mode: 'resting',
-    title: '',
-    description: '',
-    visibility: 'CLOSED',
-    group_id: null,
-    visit_channel: true,
-    errorMessage: null,
-    attributeErrors: {},
+  static defaultProps = {
+    authorType: 'USER',
+    group_id: 0,
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      mode: 'resting',
+      title: '',
+      description: '',
+      visibility: 'CLOSED',
+      group_id: props.group_id,
+      authorType: props.authorType,
+      visit_channel: true,
+      errorMessage: null,
+      attributeErrors: {},
+    };
```

Comment:
If you want to set a value in state with a props you don't need to pull it into the constructor you can just use:

```javascript
state = {
  foo: this.props.foo,
}
```

Reply:
ah, didn't know that

Reply:
will fix